exercises/guillaume_communie/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,40 @@
+#include "ShapeSorter.h"
+
+#include <iostream>
+#include <algorithm>
+
+void ShapeSorter::printMatchType(std::vector<Shape*> v,
+                                                std::string t) {
+    for (auto it = v.begin() ; it != v.end() ; it++)
+        if ((*it)->getType() == t)
+            std::cout << (*it)->print() << std::endl;
+}
+
+void ShapeSorter::printMatchSides(std::vector<Shape*> v, int n) {
+    for (auto it = v.begin() ; it != v.end() ; it++)
+        if ((*it)->getSides() == n)
+            std::cout << (*it)->print() << std::endl;
+}
+
+void ShapeSorter::printSortedByPerimeter(std::vector<Shape*> v) {
```

Comment:
You are sorting the original vector, but should do a copy; while this should print sorted, it should not modify the original vector.

Reply:
I am sorting a copy of the vector. I changed the documentation of the two functions in accordance.